llvm-external-projects/iree-compiler-api/lib/CAPI/Compiler.cpp

```diff
@@ -59,7 +64,28 @@ struct CompilerOptions {
 
 DEFINE_C_API_PTR_METHODS(IreeCompilerOptions, CompilerOptions)
 
-void ireeCompilerRegisterTargetBackends() { registerHALTargetBackends(); }
+void ireeCompilerRegisterAllDialects(MlirContext context) {
+  DialectRegistry registry;
+  registerAllDialects(registry);
```

Comment:
Would you mind namespace qualifying this (`mlir::iree_compiler::`)? There are a lot of similarly named functions and it helps to be obvious (and protects against future accidental shadowing.